# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessibleBase.cpp
# Commit: 020cd9851fea
# Full Hash: 020cd9851fea2e068e3fd84e523df155d2a3aed7
# Author: James Teh <jteh@mozilla.com>
# Date: 2023-02-03 09:16:39
# Description:
#   Bug 1814538: Don't call RemoteAccessibleBase::IsFixedPos if mCachedFields is null. r=nlapre
#   
#   RetrieveCachedBounds returns Nothing() if mCachedFields is null.
#   Since we already have a large block which is only executed if RetrieveCachedBounds returned something, just move the call to IsFixedPos inside that block.
#   
# ==============================================================================

diff -r 19576e8a0027 -r 020cd9851fea accessible/ipc/RemoteAccessibleBase.cpp
--- a/accessible/ipc/RemoteAccessibleBase.cpp	Thu Feb 02 23:18:10 2023 +0000
+++ b/accessible/ipc/RemoteAccessibleBase.cpp	Thu Feb 02 23:25:54 2023 +0000
@@ -557,6 +557,7 @@
 
 template <class Derived>
 bool RemoteAccessibleBase<Derived>::IsFixedPos() const {
+  MOZ_ASSERT(mCachedFields);
   if (auto maybePosition =
           mCachedFields->GetAttribute<RefPtr<nsAtom>>(nsGkAtoms::position)) {
     return *maybePosition == nsGkAtoms::fixed;
@@ -640,12 +641,13 @@
           bounds.MoveBy(remoteBounds.X(), remoteBounds.Y());
           Unused << remoteAcc->ApplyTransform(bounds, remoteBounds);
         }
+
+        if (remoteAcc->IsFixedPos()) {
+          encounteredFixedContainer = true;
+        }
+        // we can't just break here if we're scroll suppressed because we still
+        // need to find the top doc.
       }
-      if (remoteAcc->IsFixedPos()) {
-        encounteredFixedContainer = true;
-      }
-      // we can't just break here if we're scroll suppressed because we still
-      // need to find the top doc
       acc = acc->Parent();
     }
 
